x/shield/keeper/keeper.go

```diff
@@ -135,11 +135,7 @@ func (k Keeper) DequeueCompletedWithdrawalQueue(ctx sdk.Context) {
 		}
 		collateral.Amount = collateral.Amount.Sub(withdrawal.Amount)
 		collateral.Withdrawal = collateral.Withdrawal.Sub(withdrawal.Amount)
-		if collateral.Amount.IsZero() {
-			store.Delete(types.GetCollateralKey(pool.PoolID, collateral.Provider))
-		} else {
-			k.SetCollateral(ctx, pool, collateral.Provider, collateral)
-		}
```

Comment:
I'm pretty sure we can take this back with an additional condition (locked == 0). I think we still have to prune collaterals as much as we can 🤔

Reply:
Fixed!